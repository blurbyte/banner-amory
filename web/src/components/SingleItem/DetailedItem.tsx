import * as React from 'react';
import { Helmet } from 'react-helmet';

import { Item as ItemProps } from '@sharedTypes/Item';
import Headline from '../AnimatedHeadline';
import Layout from './Layout';
import Info from './Info';
import Details from './Details';

class DetailedItem extends React.Component<ItemProps> {
  render() {
    const {
      name,
      slug,
      description,
      gamePart,
      rank,
      price,
      acquirementMarketplace,
      acquirementQuest,
      acquirementGodstone,
      acquirementHero,
      trivia,
      statStrength,
      statWillpower,
      statArmor,
      statArmorBreak,
      talentAllRanks,
      talentStrength,
      talentWillpower,
      talentArmor,
      talentArmorBreak,
      talentExertion,
      perTurnArmor,
      perTurnWillpower,
      onRestArmor,
      onRestWillpower,
      perKillWillpower,
      aggro,
      chanceCrit,
      chanceDodge,
      chanceDivert,
      neverMiss,
      movement,
      resist,
      knockback,
      defy,
      extraRange,
      extraStrengthOnAttack,
      comments
    } = this.props;
    return (
      <>
        <Helmet>
          <title>{name}</title>
        </Helmet>
        <Headline>{name}</Headline>
        <Layout>
          <Info
            {...{
              slug,
              statStrength,
              statWillpower,
              statArmor,
              statArmorBreak,
              talentAllRanks,
              talentStrength,
              talentWillpower,
              talentArmor,
              talentArmorBreak,
              talentExertion,
              perTurnArmor,
              perTurnWillpower,
              onRestArmor,
              onRestWillpower,
              perKillWillpower,
              aggro,
              chanceCrit,
              chanceDodge,
              chanceDivert,
              neverMiss,
              movement,
              resist,
              knockback,
              defy,
              extraRange,
              extraStrengthOnAttack
            }}
          />
          <Details
            {...{
              description,
              gamePart,
              rank,
              price,
              acquirementMarketplace,
              acquirementQuest,
              acquirementGodstone,
              acquirementHero,
              trivia,
              comments
            }}
          />
        </Layout>
      </>
    );
  }
}

export default DetailedItem;
