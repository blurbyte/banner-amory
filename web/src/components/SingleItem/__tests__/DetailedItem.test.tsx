import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/test-utils';

import DetailedItem from '../DetailedItem';
import { testItem } from './mocks';

test('renders correctly', () => {
  const component = renderer.create(
    <MockedProvider mocks={[]} addTypename={false}>
      <DetailedItem {...testItem} />
    </MockedProvider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
