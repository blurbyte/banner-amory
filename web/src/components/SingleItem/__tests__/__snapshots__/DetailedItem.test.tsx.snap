// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1
    className="sc-bwzfXH cgiPFg sc-bdVaJa krEtey"
  >
    Alette's Bracelet
  </h1>,
  <div
    className="sc-htpNat fTHudk"
  >
    <div
      className="sc-gqjmRU iCmtpl"
    >
      <div
        className="sc-bxivhb hMUPsM"
      >
        <img
          className="sc-ifAKCX dKwlvE"
          src="undefinedalettes-bracelet.png"
        />
      </div>
      <div
        className="sc-VigVT cxulEs"
      >
        <svg
          viewBox="0 0 50 90"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M50.005 90L25.003 67.452 0 90V0h50.005v90z"
            fill="currentColor"
          />
        </svg>
        <div
          className="sc-jTzLTM cFrlBg"
        >
          <dl
            className="sc-EHOje cGzteA"
          >
            <dt
              className="sc-htoDjs iwPGiT sc-gzVnrw dBHQgL"
            >
              Strength
            </dt>
            <dd
              className="sc-dnqmqq iHimWC"
            >
              <span
                className="sc-gZMcBi kSuIwq"
              >
                +
                1
              </span>
            </dd>
          </dl>
          <dl
            className="sc-EHOje cGzteA"
          >
            <dt
              className="sc-htoDjs iwPGiT sc-gzVnrw dBHQgL"
            >
              Willpower
            </dt>
            <dd
              className="sc-dnqmqq iHimWC"
            >
              <span
                className="sc-gZMcBi kSuIwq"
              >
                +
                1
              </span>
            </dd>
          </dl>
          <dl
            className="sc-EHOje cGzteA"
          >
            <dt
              className="sc-htoDjs iwPGiT sc-gzVnrw dBHQgL"
            >
              Armor
            </dt>
            <dd
              className="sc-dnqmqq iHimWC"
            >
              <span
                className="sc-gZMcBi kSuIwq"
              >
                +
                1
              </span>
            </dd>
          </dl>
        </div>
      </div>
    </div>
    <div
      className="sc-cSHVUG bOtudu"
    >
      <p
        className="sc-kAzzGY iHAUGK"
      >
        Lovingly made by a hunter's wife to keep her daughter safe from harm.
      </p>
      <dl
        className="sc-EHOje eKjpuR"
      >
        <dt
          className="sc-htoDjs foKTTs sc-gzVnrw dBHQgL"
        >
          Part
        </dt>
        <dd
          className="sc-dnqmqq iXQNhl"
        >
          The Banner Saga 1
        </dd>
      </dl>
      <dl
        className="sc-EHOje eKjpuR"
      >
        <dt
          className="sc-htoDjs foKTTs sc-gzVnrw dBHQgL"
        >
          Rank
        </dt>
        <dd
          className="sc-dnqmqq bTKoEE"
        >
          1
        </dd>
      </dl>
      <dl
        className="sc-EHOje eKjpuR"
      >
        <dt
          className="sc-htoDjs foKTTs sc-gzVnrw dBHQgL"
        >
          How to get
        </dt>
        <dd
          className="sc-dnqmqq iXQNhl"
        >
          <strong
            className="sc-jzJRlG gZZLXP"
          >
            Alette
          </strong>
           starts with it
        </dd>
      </dl>
    </div>
  </div>,
]
`;
